pkg/health/report.go

```diff
@@ -0,0 +1,37 @@
+package health
+
+import (
+	"encoding/json"
+
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/pkg/config"
+	"github.com/pkg/errors"
+)
+
+// JSONReport returns a JSON representation of the diagnosis.
+func JSONReport(healthCriteria []config.Metric, diagnosis Diagnosis) (string, error) {
+	var resultsMap []map[string]interface{}
+	for i, result := range diagnosis.CheckResults {
+		criteria := healthCriteria[i]
+		report := map[string]interface{}{
+			"metricsType":   criteria.Type,
+			"threshold":     result.Threshold,
+			"actualValue":   result.ActualValue,
+			"isCriteriaMet": result.IsCriteriaMet,
+		}
+		if criteria.Type == config.LatencyMetricsCheck {
+			report["percentile"] = criteria.Percentile
+		}
```

Comment:
I also considered using the `health.Diagnosis` struct directly, but there were two drawbacks
1) We have previously decided not to include the metrics type in `health.CheckResult` since that would be repeating the same data. Plus, the percentile is also not part of that struct.
2) The `OverallDiagnosis` property is a `DiagnosisResult` (`int), so the JSON representation turns out to be a value 0..3, which is not that helpful for our users.